Build warnings on subsys/net/lib/websocket/websocket.c

### Describe the bug

when speed optimization enabled subsys/net/lib/websocket/websocket.c gives build warnings

### Regression

- [ ] This is a regression.

### Steps to reproduce

Build samples/net/sockets/websocket_client example with speed optimization enable (CONFIG_SPEED_OPTIMIZATIONS)

### Relevant log output

```shell
websocket.c:342:9:warning:'strncpy' output truncated copying between 0 and 36 bytes from a string of length 36 [-Wstringop-truncation]

343: strncpy(key_accept + key_len, WS_MAGIC, olen)
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

OS: windows
Toolchain: zephyr-sdk-0,16.8
Zephyr version: 3.7.0

### Additional Context

_No response_

--- a/subsys/net/lib/websocket/websocket.c
+++ b/subsys/net/lib/websocket/websocket.c
@@ -365,7 +365,7 @@ int websocket_connect(int sock, struct websocket_request *wreq,
 		key_len);
 
 	olen = MIN(sizeof(key_accept) - 1 - key_len, sizeof(WS_MAGIC) - 1);
-	strncpy(key_accept + key_len, WS_MAGIC, olen);
+	memcpy(key_accept + key_len, WS_MAGIC, olen);
 
 	/* This SHA-1 value is then checked when we receive the response */
 #ifdef CONFIG_MBEDTLS_PSA_CRYPTO_CLIENT

